src/ape/api/providers.py

```diff
@@ -1042,13 +1041,23 @@ def prepare_transaction(self, txn: TransactionAPI) -> TransactionAPI:
 
     def send_transaction(self, txn: TransactionAPI) -> ReceiptAPI:
         try:
-            account = self.account_manager[txn.sender]
-            if isinstance(account, ImpersonatedAccount):
+            # account = self.account_manager[txn.sender]
+            # if isinstance(account, ImpersonatedAccount):
+            #     txn_dict = txn.dict()
+            #     txn_params = cast(TxParams, txn_dict)
+            #     txn_hash = self.web3.eth.send_transaction(txn_params)
+            # else:
+            #     txn_hash = self.web3.eth.send_raw_transaction(txn.serialize_transaction())
+            if txn.signature:
+                txn_hash = self.web3.eth.send_raw_transaction(txn.serialize_transaction())
+            else:
+                if (
+                    txn.sender not in self.chain_manager.provider.web3.eth.accounts  # type: ignore # noqa
```

Comment:
I am not exactly sure why, but when we send calls as txns for the sake of getting gas and trace info on the call, the txn goes through with a sender equal to `None`. That is what led to this PR: https://github.com/ApeWorX/ape/pull/1276
It was causing issues because Ape was trying to impersonate `None`.

Perhaps there is a better way, such as setting sender on `AccountAPI.call()` if it is missing. Not sure.

But `sender` is `Optional` on a txn so we shouldn't assume it is there anyway.

Reply:
I just tried accounts[None] and test_accounts[None], as getitem has no implementation regarding None it cant impersonate it and should raise an error. How that was possible in your case?

Reply:
I introduced a new error type SenderNotFound and edited some parts so technically in send_transaction there shouldnt be any None txn.sender

Reply:
It not about impersonation, it happens when extracting gas / trace info from calls (like when `--gas`) flag is specified.  So there really isn't a sender needed.

If there is no sender, can we just send the txn like we were doing before? Else it breaks the `ape-hardhat` and `ape-foundry` plugins.

Reply:
I am testing on `ape-hardhat` and `ape-foundry`... if it is not broken, then it's fine.

Reply:
I will revert my error type as it is also neglecting contract instances and creating issues with tests. I will just send txn if there is no sender anyways and continue with other improvements if i can find other things

Reply:
Another idea I didnt try:  In the method `_send_call_as_txn`, maybe we can include the account as the sender there when it calls `call()` https://github.com/ApeWorX/ape/blob/main/src/ape/api/providers.py#L851

Reply:
It is already using account.call and for those cases i still added sender for txn if there is none.